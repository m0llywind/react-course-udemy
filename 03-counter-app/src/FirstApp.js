import React from 'react'
import PropTypes from 'prop-types'

// Functional components
const FirstApp = ({ salute, subtitle }) => {

  return (
    <>
      <h1>{ salute }</h1>
      {/* <pre>{ JSON.stringify(salute, null, 3) }</pre> */}
      <p>{ subtitle }</p>
    </>
  )
}

// Se definen los tipos de props.
// Sirve para validaciones en los componentes.
FirstApp.propTypes = {
  salute: PropTypes.string.isRequired
}

// Valores por defecto de las propiedades
FirstApp.defaultProps = {
  subtitle: 'Soy un subtitulo'
}

export default FirstApp;