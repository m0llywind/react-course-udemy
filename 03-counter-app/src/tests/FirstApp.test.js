import React from 'react';
import { render } from '@testing-library/react'
import FirstApp from '../FirstApp'
import { shallow } from 'enzyme';

describe('Tests in <FirstApp />', () => {
  // test('should show title', () => {
  //   const salute = 'Hola, Soy Laura';
  //   const { getByText } = render(<FirstApp salute={ salute } />);
  //   expect(getByText(salute)).toBeInTheDocument();
  // })

  // Usando Enzyme
  test('should show <FirstApp /> correctly', () => {
    const salute = 'Hola, Soy Laura';
    const wrapper = shallow(<FirstApp salute={salute}/>);

    // Comprueba si coincide con la ultima snapshot del componente.
    expect(wrapper).toMatchSnapshot();
  })

  test('should show the subtitle', () => {
    const salute = 'Hola, Soy Laura';
    const subtitle = 'Soy un subtitulo';

    const wrapper = shallow(
      <FirstApp 
        salute={salute}
        subtitle={subtitle} 
      />
    );

    const text = wrapper.find('p').text();

    expect(text).toBe(subtitle);
  })
})
