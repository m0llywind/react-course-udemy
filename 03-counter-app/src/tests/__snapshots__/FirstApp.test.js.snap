// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should show <FirstApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola, Soy Laura
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
